gips/0080-gip-direct-indexing-payments.md

```diff
@@ -0,0 +1,147 @@
+- When the Gateway wants an Indexer to perform indexing work, it sends a signed **Voucher**. The Voucher specifies the max cost per period, the agreed price per gas/unit of work, and the details of the work to do (i.e. which subgraph).
+- The Indexer can post that Voucher onchain, which results in accepting the **Agreement**, the terms for which are encoded in the Voucher. The Voucher specifies that the Subgraph Service contract is authorized to release the payment; the Subgraph Service contract also specifies the stake-to-fees ratio. From the moment the voucher is posted onchain, the Indexer is guaranteed to be able to collect payment if they post the POI (indexing up to the current Epoch) within the specified time and meeting all necessary conditions (e.g. provisioned stake).
+- The Gateway doesn’t need to sign anything else to release further payments, they just need to keep the Escrow topped up. The Indexer can stop working if they don’t see sufficient funds in the Escrow. This ensures that Gateways cannot avoid payment after the Indexer has done the work.
+- The Subgraph Service releases payments conditional on Indexers posting POIs - this ensures that Indexers cannot collect payment without doing the work (and if they present an invalid POI, they can be slashed). The voucher release contract can ensure that only up to the maximum amount is released, but the Subgraph Service can release the exact amount based on the agreed pricing (e.g. price per gas times the reported indexing gas used).
+
+**This voucher release mechanism is generic and can be used for other data services:** it validates that the caller is the approved data service contract, that the token amount is within the maximum value specified by the voucher per period, and that the right amount of time has passed. The data service specific aspects e.g. POI validity, gas/pricing, etc. are defined at the data service contract level. Moreover, this contract can be immutable, so payers are protected even if the data service contract is upgraded.
+
+It is important to note that we cannot expect Indexers to take on risk, e.g. by agreeing to a fixed price for a specific subgraph (instead of per unit of work/gas). The cost of indexing each subgraph is unknown until the subgraph has been indexed, so if the price was fixed a rational Indexer would simply stop indexing when the subgraph becomes unprofitable, forcing the payer to start over. Tying payments as much as possible to the verifiable cost of performing the work is the best way to prevent Denial of Service scenarios, just like with gas in Ethereum.
+
+### Terms and Definitions
+
+**Curation**: Curation requires Consumers to stake a certain amount of GRT per subgraph to signal Indexers to provide indexing services.  However, because the strength of the signal depended upon the strength of other signals using GRT, which also had variability, this became a hard mechanism for developers with many subgraphs.
```

Comment:
```suggestion
**Curation**: Curation requires Consumers to "signal" a certain amount of GRT (by depositing it to a smart contract) per subgraph to signal Indexers to provide indexing services, as curation signal affects the amount of Indexing Rewards.  However, because the strength of the signal depends upon the amount of curation on other subgraphs, which has variability, this can be a hard mechanism for developers with many subgraphs.
```